ckanext/datagovuk/lib/cli.py

```diff
@@ -39,4 +51,238 @@ def command(self):
                 create_reset_key(user)
                 user.save()
 
-                writer.writerow([user.id, user.email, user.reset_key])
\ No newline at end of file
+                writer.writerow([user.id, user.email, user.reset_key])
+
+
+class CreateDGUTestDataCommand(CkanCommand):
+    '''Creates the test data so that CKAN API functional tests can run
+
+        Set your environment varibales:
+            CKAN_INI - location of CKAN ini file 
+            CKAN_TEST_SYSADMIN_NAME and CKAN_TEST_SYSADMIN_PASSWORD
+
+        paster --plugin=ckanext-datagovuk create_dgu_test_data -c $CKAN_INI
+    '''
+    summary = __doc__.split('\n')[0]
+    usage = '\n'.join(__doc__.split('\n')[1:])
+    max_args = None
+    min_args = None
+
+    def add_empty_field(dataset_id, fieldname, model):
+        field = model.PackageExtra(package_id=dataset_id, key=fieldname, value="")
+        model.Session.add(field)
+
+    def command(self):
+        if not [i for i in os.environ if i in ['CKAN_TEST_SYSADMIN_NAME', 'CKAN_TEST_SYSADMIN_PASSWORD', 'CKAN_INI']]:
```

Comment:
I'm not sure this logic does what the error message suggests. I think you want:

```
if not all(os.environ.get(i) for i in ('CKAN_TEST_SYSADMIN_NAME', 'CKAN_TEST_SYSADMIN_PASSWORD', 'CKAN_INI')):
```

The `.get(i)` so that empty strings fall through and count as falsey.

Reply:
I've tested my code and it appears to work but I think your code is clearer so will implement that

Reply:
Well from the looks of it what yours does is pass if _any_ of those three is set. Unless ... that's what you wanted? The error message is ambiguous.